libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -72,6 +74,41 @@ export class CartService {
     private paymentIntentManager: PaymentIntentManager
   ) {}
 
+  /**
+   * Should be used to wrap any method that mutates an existing cart.
+   * This transforms errors into a cart error reason ID and finalizes the cart with an error.
+   */
+  private async wrapWithCartCatch<T>(cartId: string, method: () => Promise<T>) {
```

Comment:
Two comments here:
1. I think this should be converted to a decorator method. It inherently does not modify the original return value, but injects an "on failure" clause, which feels like it nicely matches the try/catch decorator pattern.
2. The function's name does not describe the impact of the function. Something like `failCartOnReject` might be clearer!

Reply:
It looks like each of the calls to `wrapWithCartCatch` wraps the entirety of the affected method's inner logic (part of why it feels very `decorator`y to me)

Reply:
The implementation follows the described behavior in the ticket that was pointed/sliced -- given that, I'd prefer to move forward with the implementation as it is here. I'll definitely respond to these comments, but just want to make it clear I'd like to merge it as the ticket describes.

1. Moving to a decorator:
- Changing this to a decorator would inhibit my ability to use it here: https://github.com/mozilla/fxa/pull/18150/files#diff-20bf2f21059e7204e19eaa034be993bf62b2f4356de22faed1981b6bf4d37f99L256-L277
- Changing to a decorator would make it difficult to pull the `cartId` out of the method. Typings for decorators are generally rough, and given that it inherently relies on cartId from the provided method, we'd either need a context attached similar to type-cacheable, or assume the location in the originating function.
2. The name seems clear enough to me, but I'm not opposed to changing it. Given that Reino was the original creator of the ticket, lets defer to him if that's alright with you.